src/main/java/com/epam/service/UserServiceImpl.java

```diff
@@ -0,0 +1,65 @@
+package com.epam.service;
+
+import com.epam.model.User;
+import com.epam.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.util.Assert;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class UserServiceImpl implements UserService {
```

Comment:
please create unit tests for this service(mock all dependencies - UserRepository is this case)

Reply:
Done.